.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,237 @@
+        run: pip install boto3
+      - name: Get a snapshot of metrics and commit them to the repository
+        run: |
+          python3 .github/scripts/performance-tests/produce_metric_widget_images.py \
+            --logs-namespace ${{ env.LOGS_NAMESPACE }} \
+            --metrics-period ${{ env.HOSTMETRICS_INTERVAL_SECS }} \
+            --num-of-cpus ${{ env.NUM_OF_CPUS }} \
+            --app-process-command-line-dimension-value "${{ env.APP_PROCESS_COMMAND_LINE_DIMENSION_VALUE }}" \
+            --target-sha ${{ env.TARGET_SHA }} \
+            --github-run-id ${GITHUB_RUN_ID} \
+            --test-duration-minutes ${{ env.TEST_DURATION_MINUTES }} \
+            --cpu-load-threshold ${{ env.CPU_LOAD_THRESHOLD }} \
+            --total-memory-threshold ${{ env.TOTAL_MEMORY_THRESHOLD }} \
+            --app-platform ${{ matrix.app-platform }} \
+            --instrumentation-type ${{ matrix.instrumentation-type }} \
+            --max-benchmarks-to-keep ${{ env.MAX_BENCHMARKS_TO_KEEP }} \
+            --github-repository ${GITHUB_REPOSITORY}
+          echo "::warning::Checkout Snapshots at this link: https://github.com/${GITHUB_REPOSITORY}/blob/gh-pages/soak-tests/snapshots/commits/${{ env.TARGET_SHA }}/runs/${GITHUB_RUN_ID}/${{ matrix.app-platform }}";
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com";
+          git config user.name "GitHub Actions";
+          git fetch;
+          git checkout gh-pages;
+          git add soak-tests/snapshots/commits;
+          git commit -m "Soak Test Snapshots from ${{ env.TARGET_SHA }} - ${GITHUB_RUN_ID}";
+          git push;
+          git checkout main;
+      - name: Prepare Performance Test results as JSON output
+        run: python3 .github/scripts/performance-tests/get-metric-data/produce_performance_test_results.py
+          --logs-namespace ${{ env.LOGS_NAMESPACE }}
+          --metrics-period ${{ env.HOSTMETRICS_INTERVAL_SECS }}
+          --num-of-cpus ${{ env.NUM_OF_CPUS }}
+          --app-process-command-line-dimension-value "${{ env.APP_PROCESS_COMMAND_LINE_DIMENSION_VALUE }}"
+          --target-sha ${{ env.TARGET_SHA }}
+          --github-run-id ${GITHUB_RUN_ID}
+          --test-duration-minutes ${{ env.TEST_DURATION_MINUTES }}
+      - name: Do we already have Performance Test graph results for this commit?
+        continue-on-error: true
+        id: check-already-have-performance-results
+        run: |
+          git checkout gh-pages;
+          HAS_RESULTS_ALREADY=$(
+            sed 's/window.BENCHMARK_DATA = //' soak-tests/per-commit-overall-results/data.js |
+            jq "
+              .entries |
+              .\"Soak Test Results - sample-app-${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}\" |
+              any(.commit.id == \"${{ env.TARGET_SHA }}\")
+            " || echo false
+          );
+          git checkout main;
+          [[ $HAS_RESULTS_ALREADY == true ]]
```

Comment:
Instead of having this step fail, why not emit a variable that can be checked? We do that in the collector with https://github.com/aws-observability/aws-otel-collector/blob/main/tools/workflow/is-commit-benchmarked.sh, which was based on the `aws-otel-java-instrumentation` workflow you wrote.

Reply:
Is there a reason you still run the tests even if it's been done before?

Reply:
> Is there a reason you still run the tests even if it's been done before?

We run the tests again because the Docker Build makes sure we test the latest from upstream.

But on this line specifically we are just checking if we should publish results for it. We have a graph of results per commit, but even though we safe the results from every run on the same commit, that graph can only keep 1 result. So we just decided to keep the 1st run instead of overwriting it each time. You can see the graph here: https://aws-observability.github.io/aws-otel-python/soak-tests/per-commit-overall-results/index.html

Keeping the different runs is useful because the templates (also in this PR) use the `env.GITHUB_RUN_ID` variable to always get the latest graphs in an issue! The workflow will automatically update pending Soak Test Performance Issues on the GitHub repo 🙂